test/jdk/javax/swing/plaf/metal/MetalBorders/ScaledMetalBorderTest.java

```diff
@@ -0,0 +1,351 @@
+                || borderDirection.equals("BOTTOM");
+
+        robot.mouseMove(x, y);
+        for (int i = start; i < stop; i++) {
+            int locX = isVertical ? i : (windowLoc.x + MIDPOINT);
+            int locY = isHorizontal ? i : (windowLoc.y + MIDPOINT);
+            if (Color.RED.equals(robot.getPixelColor(locX, locY))) {
+                errorLog.append("At uiScale: " + uiScale
+                        + ", Red background color detected at "
+                        + borderDirection + " border.\n");
+                break;
+            }
+        }
+        robot.delay(100);
+    }
+
+    private static void checkCorners(String cornerLocation) {
+        int x, y;
+
+        switch (cornerLocation) {
+            case "TOP_LEFT" -> {
+                x = windowLoc.x;
+                y = windowLoc.y;
+            }
+            case "TOP_RIGHT" -> {
+                x = windowMaxX;
+                y = windowLoc.y;
+            }
+            case "BOTTOM_RIGHT" -> {
+                x = windowMaxX;
+                y = windowMaxY;
+            }
+            case "BOTTOM_LEFT" -> {
+                x = windowLoc.x;
+                y = windowMaxY;
+            }
+            default -> throw new IllegalStateException("Unexpected value: "
+                    + cornerLocation);
+        }
+
+        boolean isTop = cornerLocation.equals("TOP_LEFT")
+                || cornerLocation.equals("TOP_RIGHT");
+        boolean isLeft = cornerLocation.equals("TOP_LEFT")
+                || cornerLocation.equals("BOTTOM_LEFT");
+
+        robot.mouseMove(x, y);
+        for (int i = 0; i < BORDER_THICKNESS - 1; i++) {
+            int locX = isLeft ? (x + i) : (x - i);
+            int locY = isTop ? (y + i) : (y - i);
+            if (Color.RED.equals(robot.getPixelColor(locX, locY))) {
+                errorLog.append("At uiScale: " + uiScale + ", Red background color"
+                        + " detected at " + cornerLocation + " corner.\n");
+                break;
+            }
+        }
+        robot.delay(100);
+    }
+
+    private static void createFrame() {
+        jFrame = new JFrame("Frame with Metal Border");
+        jFrame.setSize(SIZE, SIZE);
+        jFrame.setBackground(Color.RED);
+        jFrame.getContentPane().setBackground(Color.RED);
+        jFrame.setLayout(new GridBagLayout());
+        jFrame.getContentPane().add(scale);
+        jFrame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+        jFrame.setLocation(150, 150);
+        jFrame.setVisible(true);
+    }
+
+    private static void createDialog() {
+        jDialog = new JDialog((Frame) null , "Dialog with Metal Border");
+        jDialog.setSize(SIZE, SIZE);
+        jDialog.setBackground(Color.RED);
+        jDialog.getContentPane().setBackground(Color.RED);
+        jDialog.setLayout(new GridBagLayout());
+        jDialog.getContentPane().add(scale);
+        jDialog.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
```

Comment:
```suggestion
        jDialog.setDefaultCloseOperation(JDialog.DISPOSE_ON_CLOSE);
```
Would be more logical.